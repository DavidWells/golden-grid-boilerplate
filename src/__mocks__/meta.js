// eslint-disable-next-line import/prefer-default-export
export const meta = {
  lang: 'en',
  path: './',
  charset: 'UTF-8',
  title: 'Golden canon grid website boilerplate',
  metaDesc: 'Georgina Alson portfolio golden canon grid website boilerplate created with css grid',
  template: 'base-template',
  // String value
  home: {
    name: 'Georgina',
    surname: 'Alson',
    profession: 'YUNG MODEL 2020',
    menuLink_1: {
      link: '/',
      name: 'CLIENTS'
    },
    menuLink_2: {
      link: '/',
      name: 'NEWS'
    },
    profileLink: {
      link: '/',
      name: 'VIEW PROFILE'
    }
  },
  // Navigation menu
  navigation: [
    {
      title: 'Home',
      link: '/index.html'
    },
    {
      title: 'Profile',
      link: '/profile.html'
    },
    {
      title: 'Contact',
      link: '/contact.html'
    },
    {
      title: 'Client',
      link: '/client.html'
    },
    {
      title: 'News',
      link: '/news.html'
    },
    {
      title: 'Single Blog',
      link: '/single-blog.html'
    },
    {
      title: 'About',
      link: '/about.html'
    }
  ],
  social: [
    {
      id: '#facebook',
      title: 'facebook'
    },
    {
      id: '#instagram',
      title: 'instagram'
    },
    {
      id: '#youtube',
      title: 'youtube'
    },
    {
      id: '#twitter',
      title: 'twitter'
    }
  ]
};
