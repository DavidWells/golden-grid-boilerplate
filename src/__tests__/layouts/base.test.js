import 'cross-fetch/polyfill';
import pug from 'pug';
import path from 'path';
import chalk from 'chalk';
import * as mocks from '../../__mocks__';

const templatePath = path.resolve(__dirname, '../../layouts/base.pug');
const renderPug = (data) => pug.renderFile(templatePath, data);

describe(`Testing [${chalk.yellowBright('base')}] template`, () => {
  it('Should render and match snapshot', () => {
    expect(renderPug(mocks.meta)).toMatchSnapshot();
  });
});
