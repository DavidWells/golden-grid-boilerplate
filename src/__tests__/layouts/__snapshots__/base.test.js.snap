// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing [[93mbase[39m] template Should render and match snapshot 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><title>Portfolio website (golden grid canon) boilerplate</title><meta name=\\"description\\" content=\\"Example portfolio website made using postcss and pug, could also be used as a boilerplate\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0, shrink-to-fit=no\\"><meta name=\\"theme-color\\" content=\\"#ffffff\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"font-src 'self' 'unsafe-inline' 'unsafe-eval' https://fonts.gstatic.com;\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"script-src 'self' 'unsafe-inline' 'unsafe-eval' http://www.google.com https://fonts.googleapis.com;\\"><meta http-equiv=\\"Content-Security-Policy\\" content=\\"style-src 'self' 'unsafe-inline' 'unsafe-eval' https://fonts.googleapis.com;\\"><!-- Microsoft --><meta name=\\"msapplication-TileColor\\" content=\\"#ffffff\\"><meta name=\\"msapplication-TileImage\\" content=\\"./favicon-144x144.png\\"><!-- Apple --><link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"./favicon-180x180.png\\"><link rel=\\"preconnect\\" href=\\"https://fonts.gstatic.com/\\" crossorigin><link rel=\\"manifest\\" href=\\"./manifest.json\\"><link rel=\\"icon\\" type=\\"image/png\\" href=\\"./favicon-32x32.png\\" sizes=\\"32x32\\"><link rel=\\"icon\\" type=\\"image/png\\" href=\\"./favicon-16x16.png\\" sizes=\\"16x16\\"><link href=\\"favicon.ico\\" rel=\\"shortcut icon\\" type=\\"image/x-icon\\"><base href=\\"/\\"></head><body><!-- +radioSwitch('select theme', 'theme-selector', themeSwitch)--><header class=\\"header\\"><!--a.header-logo-link(href=\\"/\\")--><!--  +icon('#pug', 'logo')--><!----><!--div.theme-switch-wrapper--><!--  +checkBoxSwitch('theme-selector', 'Dark mode')--><!----><!--nav.menu--><!--  +menu(navMenu, currentNav)--></header><main class=\\"base-template\\"><section></section></main><footer class=\\"footer\\"><div>Footer content</div><p>Copyright</p></footer><footer class=\\"app-bar-container\\"><p>mobile app-bar content</p></footer></body><noscript><p>You need to enable JavaScript to run this app.</p></noscript></html>"`;
